samples/aspnetcore/blazor/FlightFinder/FlightFinder.Client/Flights.cshtml

```diff
@@ -0,0 +1,33 @@
+﻿@page "/"
+@layout MainLayout
+@inject AppState state
+
+<div id="search-and-results-area">
+    <SearchView SearchInProgress="state.SearchInProgress"
+                Sort="SortOptions"
+                Results="state.FlightSearchResults">
+        <SearchContent>
+            <FlightSearch OnSearch=state.SearchFlights />
+        </SearchContent>
+        <SearchResultItemContent Context="result">
+            <FlightSearchResult Item="result" OnAddItinerary="state.AddToShortlist" />
+        </SearchResultItemContent>
+    </SearchView>
+</div>
+<div id="selections-area" class="p-3 pl-4">
+    <Shortlist Items=state.Shortlist />
+</div>
+
+@functions
+{
+    static readonly SearchView<FlightItinerary>.SortOrder[] SortOptions = new SearchView<FlightItinerary>.SortOrder[]
+    {
+        new SearchView<FlightItinerary>.SortOrder() { DisplayName = "Cheapest", Sort = r => r.OrderBy(i => i.Price), },
+        new SearchView<FlightItinerary>.SortOrder() { DisplayName = "Quickest", Sort = r => r.OrderBy(i => i.TotalDurationHours), },
+    };
```

Comment:
Considering putting this initialization elsewhere. Thoughts?

Reply:
TBH although this way of factoring components is great for demoing templated components, and I'm glad you've done it for the demos this week, I suspect it's slightly artificial to have `SearchView` as an abstracted thing (it makes the factoring rather complex for both cars and flights, instead of having the "search" and "results" parts for each separately inside `CarRentals.cshtml` and `Flights.cshtml`).

Personally I think the structure you have in this PR will be excellent for the immediate demo requirements so I'd use this as-is for those demos.